cosmo_tester/test_suites/test_manager_upgrade/manager_upgrade_test.py

```diff
@@ -0,0 +1,322 @@
+
+        self.manager_private_ip = self._load_private_ip_from_env(workdir)
+
+        # TODO: why is this needed?
+        self.manager_cfy.use(management_ip=self.upgrade_manager_ip)
+
+    def post_bootstrap_checks(self):
+        self.rest_client.blueprints.list()
+
+    def deploy_hello_world(self, prefix=''):
+        blueprint_id = prefix + self.test_id
+        deployment_id = prefix + self.test_id
+        hello_repo_dir = tempfile.mkdtemp(prefix='manager-upgrade-')
+        hello_repo_path = clone(
+            'https://github.com/cloudify-cosmo/'
+            'cloudify-hello-world-example.git',
+            hello_repo_dir
+        )
+
+        hello_blueprint_path = hello_repo_path / 'blueprint.yaml'
+        self.manager_cfy.upload_blueprint(blueprint_id, hello_blueprint_path)
+        self.addCleanup(self.manager_cfy.delete_blueprint, blueprint_id)
+
+        # TODO if we separate base/openstack test case, this also needs to be
+        # outside
+        inputs = {
+            'agent_user': self.env.cloudify_agent_user,
+            'image': self.env.ubuntu_trusty_image_name,
+            'flavor': self.env.flavor_name
+        }
+        self.manager_cfy.create_deployment(blueprint_id, deployment_id,
+                                           inputs=inputs)
+        self.addCleanup(self.manager_cfy.delete_deployment, deployment_id)
+
+        # TODO uninstall in cleanup (separate method with check?)
+        self.manager_cfy.execute_install(deployment_id=deployment_id)
+        return deployment_id
+
+    def get_upgrade_blueprint(self):
+        repo_dir = tempfile.mkdtemp(prefix='manager-upgrade-')
+        self.addCleanup(shutil.rmtree, repo_dir)
+        upgrade_blueprint_path = clone(UPGRADE_REPO_URL,
+                                       repo_dir,
+                                       branch=UPGRADE_BRANCH)
+
+        # TODO: this uses a simple blueprint, is that right?
+        yaml_path = upgrade_blueprint_path / 'simple-manager-blueprint.yaml'
+        with YamlPatcher(yaml_path) as patch:
+            patch.set_value(
+                ('node_templates.elasticsearch.properties'
+                 '.use_existing_on_upgrade'),
+                False)
+        return yaml_path
+
+    def upgrade_manager(self, upgrade_blueprint):
+
+        upgrade_inputs = {
+            'private_ip': self.manager_private_ip,
+            'public_ip': self.upgrade_manager_ip,
+            'ssh_key_filename': self.manager_inputs['ssh_key_filename'],
+            'ssh_user': self.manager_inputs['ssh_user'],
+            'elasticsearch_endpoint_port': 9900
+
+        }
+        upgrade_inputs_file = self.manager_cfy._get_inputs_in_temp_file(
+            upgrade_inputs, self._testMethodName)
+
+        with self.manager_cfy.maintenance_mode():
+            self.manager_cfy.upgrade_manager(
+                blueprint_path=upgrade_blueprint,
+                inputs_file=upgrade_inputs_file)
+
+    def post_upgrade_checks(self, deployment_id):
```

Comment:
This method is the same as 'post_rollback_checks'. I would replace the name to 'assert_es_port_open' or something more in the lines of that..